CONTRIBUTING.md

```diff
@@ -1,29 +1,123 @@
+> and [#7](https://cla.developers.google.com/about/google-corporate#:~:text=Should%20You%20wish%20to%20submit%20work%20that%20is%20not%20Your%20original%20creation%2C%20You%20may%20submit%20it%20to%20Google%20separately)
+> in the CLA. Any code that you contribute to this project must be **your**
+> original creation. Code generated by artificial intelligence tools **does
+> not** qualify as your original creation.
+
+### Review our community guidelines
+
+We have a [code of conduct](CODE_OF_CONDUCT.md) to make the ReCirq project an open
+and welcoming community environment. Please make sure to read and abide by the
+code of conduct.
+
+## Contribution process
 
 All submissions, including submissions by project members, require review. We
-use GitHub pull requests for this purpose. Consult
-[GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
-information on using pull requests.
+use the tools provided by GitHub for [pull
+requests](https://help.github.com/articles/about-pull-requests/) for this
+purpose. The preferred manner for submitting pull requests is to fork the ReCirq
+[repository](https://github.com/quantumlib/ReCirq), create a [git
+branch](https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell) in
+this fork to do your work, and when ready, create a pull request from this
+branch to the main ReCirq repository. The subsections below describe the process
+in more detail.
+
+Pleae make sure to follow the [Google Style
+Guides](https://google.github.io/styleguide/) in your code, particularly the
+[style guide for Python](https://google.github.io/styleguide/pyguide.html).
+
+### Repository forks
+
+1.  Fork the ReCirq repository (you can use the _Fork_ button in upper right
+    corner of the [repository page](https://github.com/quantumlib/ReCirq)).
+    Forking creates a new GitHub repo at the location
+    `https://github.com/USERNAME/ReCirq`, where `USERNAME` is your GitHub user
+    name.
+
+1.  Clone (using `git clone`) or otherwise download your forked repository to
+    your local computer, so that you have a local copy where you can do your
+    development work using your preferred editor and development tools.
+
+1.  Check out the `main` branch and create a new git branch from `main`:
+
+    ```shell
+    git checkout main -b YOUR_BRANCH_NAME
+    ```
+
+    where `YOUR_BRANCH_NAME` is the name of your new branch.
+
+### Development and testing
+
+Do your work and `git commit` your changes to your branch as needed.
+
+We use several tools to test code and perform other activities such as checking
+formatting against the style guidelines. You can run those tools locally during
+development. Wrapper scripts are located in the [`check/`](./check/)
+subdirectory to simplify running the tools.
+
+*   Run `check/pytest` to run the Pytest suite
+*   Run `check/mypy` to run the Mypy type checker
+*   Run `check/pylint` to run the Pylint code linter
```

Comment:
These do not exist in ReCirq.  Please update to reflect the actual testing tools available, the jobs in [pythonpackage.yml](https://github.com/quantumlib/ReCirq/blob/a8a5791c17df4592dd63a7b50ea67a447e7e5c96/.github/workflows/pythonpackage.yml#L45-L59) could be of some guidance.  Note the CI jobs are in a bit wonky state, for example, the flake8 is run as a soft check and reports most errors as warnings.

Reply:
You're right. Looking at the situation now, the tools that are being used need to be installed, but they're not in the requirements file. This is going to need a little bit more to make it user-friendly, so for the time being, I'm going to remove the explanations of the tools and just leave the first sentence in the "Development and testing" section, and open an issue about improving the tool setup.